sound/soc/sof/ipc4-topology.c

```diff
@@ -1045,6 +1058,37 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	if (ret < 0)
 		return ret;
 
+	switch (swidget->id) {
+	case snd_soc_dapm_dai_in:
+	case snd_soc_dapm_dai_out:
+	{
+		/*
+		 * Only SOF_DAI_INTEL_ALH needs copier_data to set blob.
+		 * That's why only ALH dai's blob is set after sof_ipc4_init_audio_fmt
+		 */
+		if (ipc4_copier->dai_type == SOF_DAI_INTEL_ALH) {
+			struct sof_ipc4_alh_configuration_blob *blob;
+			u32 ch_map;
+			int i;
+
+			blob = (struct sof_ipc4_alh_configuration_blob *)ipc4_copier->copier_config;
+			/* TODO: add aggregation mode support */
+			blob->alh_cfg.count = 1;
+			blob->alh_cfg.mapping[0].alh_id = copier_data->gtw_cfg.node_id;
+			blob->gw_attr.lp_buffer_alloc = 0;
+			copier_data->gtw_cfg.config_length = sizeof(*blob) >> 2;
+
+			/* Get channel_mask from ch_map */
+			ch_map = copier_data->base_config.audio_fmt.ch_map;
+			for (i = 0; ch_map > 0; i++) {
```

Comment:
`u32` tends to be always bigger than 0 when it is not 0, `for (i = 0; ch_map; i++)` or if you really want to explicit `for (i = 0; ch_map != 0; i++)`

Reply:
> `u32` tends to be always bigger than 0 when it is not 0, `for (i = 0; ch_map; i++)` or if you really want to explicit `for (i = 0; ch_map != 0; i++)`

Thanks, I will send a fixup commit on https://github.com/thesofproject/linux/pull/3589